### PR TITLE
Remove all non-BOM entries from dependency management

### PR DESCRIPTION
**Pull request by awood**
Having non-BOM entries in dependency management results in transitive dependency exclusions not working properly.  We achieve the intended effect of dependency management (uniform versions of a library across all sub-projects) with the dependencies.gradle file anyway.

Attn @kahowell

### CODE REVIEW

**Comment by lindseyburnett** [comment]
related: https://github.com/RedHatInsights/rhsm-subscriptions/pull/2423/files#diff-81ca6ede6d5ce5bc9718e90eed7b8887c0b4ae7980ca0a8789d96075db246461R10

File: buildSrc/src/main/groovy/swatch.spring-boot-dependencies-conventions.gradle [hunk 1]
```diff
@@ -14,29 +14,10 @@ dependencies {
     testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine"
 }
 
-// Define all versions here so that all projects will use the same versions.
 dependencyManagement {
     imports {
         // See https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies to
         // get listings of the contents of this BOM
         mavenBom libraries["spring-boot-dependencies"]
```
Comment (kahowell) [inline buildSrc/src/main/groovy/swatch.spring-boot-dependencies-conventions.gradle:21]: I'd be interested to go ahead and try moving this dep to the `dependencies` section, using `enforcedPlatform` and dropping the spring dependency management plugin; if it all works out, let's just go ahead and do that here.

**Review by kahowell — CHANGES_REQUESTED** [review]

**Review by kahowell — CHANGES_REQUESTED** [review]
- [ ] `platform` was intentional for the libraries (just revert d4f0801) - because we don't want to force the spring boot services to use quarkus provided versions.
- [ ] We can remove `io.spring.gradle:dependency-management-plugin` from dependencies.gradle now.

**Comment by awood** [comment]
@kahowell Thanks for the review.  The issues you caught should be fixed now.  Good to know about the "platform was intentional for the libraries" bit.  I added a commit with comments explaining that.

**Review by kahowell — APPROVED** [review]

**Pull request merged**